### DISCUSSION TITLE
Using JavaScript functions from Rust

### DISCUSSION THREAD

**Discussion by hikmetli** [wasm-bindgen/wasm-bindgen#3860] [Q&A] [answered]
Hello, 
I want to use functions that defined from a javascript file with rust. Do i need to run a server for this? Otherwise what should i do to use functions?
Thanks.

**Comment by daxpedda** [comment, marked as answer]
I'm not sure I understand the question right, but `wasm-bindgen` is designed to be used with Rust compiled to `wasm32-unknown-unknown` and run in a JavaScript environment. It is not designed to somehow run JavaScript from a native (e.g. x86) environment.

If you are running in a JavaScript environment (e.g. browser, NodeJS), you can link to JS functions and run that. See the [`wasm-bindgen` book](https://rustwasm.github.io/docs/wasm-bindgen/examples/import-js.html) for more details.

**Reply by hikmetli** [reply]
Ok I got my answer, thank you
